src/preferences/waveformsettings.h

```diff
@@ -15,6 +15,16 @@ class WaveformSettings {
                 ConfigKey("[Library]", "EnableWaveformCaching"), enabled);
     }
 
+    bool waveformGenerateWithAnalysisEnabled() const {
+        return m_pConfig->getValue<bool>(
+                ConfigKey("[Library]", "GenerateWaveformsWithAnalysis"), true);
```

Comment:
The naming is inconsistent in various aspects. The new option in the config file uses a different wording than the corresponding key in the config file: "waveform generate" (singular) vs. "generate waveforms" (plural) and ordering of words. And it also differs from the naming of related config options, namely "EnableWaveformCaching" -> "waveform caching enabled".

I would propose to follow the existing naming scheme for related options. Even if we don't like the current scheme, at least it is easier to remember if it is consistent.